tests/foreman/api/test_role.py

```diff
@@ -100,7 +100,48 @@ def test_positive_update(self):
 class CannedRoleTestCases(APITestCase):
     """Implements Canned Roles tests from API"""
 
-    @stubbed()
+    def create_org_admin(self, name=None, orgs=None, locs=None):
+        """Helper function to create org admin role for particular
+        organizations and locations by cloning 'Organization admin' role.
+
+        :param str name: The name of cloned Org Admin role
+        :param list orgs: The list of organizations for which the org admin is
+            being created
+        :param locs: The list of locations for which the org admin is
+            being created
+        :return dict: This function returns dict representation of cloned role
+            data returned from 'clone' function
+        """
+        name = gen_string('alpha') if not name else name
+        orgs = [] if not orgs else orgs
+        locs = [] if not locs else locs
+        default_org_admin = entities.Role().search(
+            query={'search': u'name="Organization admin"'})
+        org_admin = entities.Role(id=default_org_admin[0].id).clone(
+            data={
+                'role': {
+                    'name': name,
+                    'organization_ids': orgs,
+                    'location_ids': locs
+                }
+            }
+        )
+        self.assertEqual(org_admin['name'], name)
+        return org_admin
+
+    @classmethod
+    def setUpClass(cls):
+        """Creates two orgs and locations that will be used by all the
+        underlying tests
+        """
+        super(CannedRoleTestCases, cls).setUpClass()
+        # These two will be used as role taxonomies
+        cls.org1 = entities.Organization(name=gen_string('alpha')).create()
+        cls.loc1 = entities.Location(name=gen_string('alpha')).create()
+        # These two will be used as filter taxonomies
+        cls.org2 = entities.Organization(name=gen_string('alpha')).create()
+        cls.loc2 = entities.Location(name=gen_string('alpha')).create()
```

Comment:
nitpick: no need to provide names for orgs and locs explicitly, nailgun will generate them